dashboard/src/components/ApplicationStatus/ApplicationStatus.tsx

```diff
@@ -32,7 +53,68 @@ class ApplicationStatus extends React.Component<IApplicationStatusProps> {
     this.props.closeWatches();
   }
 
+  public componentDidUpdate(prevProps: IApplicationStatusProps) {
+    if (prevProps !== this.props) {
+      const { deployments, statefulsets, daemonsets } = this.props;
+      let totalPods = 0;
+      let readyPods = 0;
+      let workloads: IWorkload[] = [];
+      deployments.forEach(d => {
+        if (d.item) {
+          const status: IDeploymentStatus = d.item.status;
+          if (status.availableReplicas) {
+            readyPods += status.availableReplicas;
+          }
+          if (status.replicas) {
+            totalPods += status.replicas;
+          }
+          workloads = workloads.concat({
+            name: d.item.metadata.name,
+            replicas: status.replicas || 0,
+            readyReplicas: status.availableReplicas || 0,
+          });
+        }
+      });
+      statefulsets.forEach(d => {
+        if (d.item) {
+          const status: IStatefulsetStatus = d.item.status;
+          if (status.readyReplicas) {
+            readyPods += status.readyReplicas;
+          }
+          if (status.replicas) {
+            totalPods += status.replicas;
+          }
+          workloads = workloads.concat({
+            name: d.item.metadata.name,
+            replicas: status.replicas || 0,
+            readyReplicas: status.readyReplicas || 0,
+          });
+        }
+      });
+      daemonsets.forEach(d => {
+        if (d.item) {
+          const status: IDaemonsetStatus = d.item.status;
+          if (status.numberReady) {
+            readyPods += status.numberReady;
+          }
+          if (status.currentNumberScheduled) {
+            totalPods += status.currentNumberScheduled;
+          }
+          workloads = workloads.concat({
+            name: d.item.metadata.name,
+            replicas: status.currentNumberScheduled || 0,
+            readyReplicas: status.numberReady || 0,
+          });
+        }
+      });
```

Comment:
Looks like the above 3 loops could be a single function call `readyPods, totalPods, workloads = collectResources(daemonsets, readyKey='numberReady', totalKey='currentNumberScheduled')` or similar, and just adding them externally. But fine as is too.